rotkehlchen/db/upgrades/v35_v36.py

```diff
@@ -616,6 +587,28 @@ def _fix_eth2_pnl_genesis(write_cursor: 'DBCursor') -> None:
         )
 
 
+def _reset_decoded_events(write_cursor: 'DBCursor') -> None:
+    write_cursor.execute('SELECT tx_hash from evm_transactions')
+    tx_hashes = [x[0] for x in write_cursor]
+    write_cursor.execute(
+        'SELECT parent_identifier FROM history_events_mappings WHERE name=? AND value=?',
+        (HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED),
+    )
+    customized_event_ids = [x[0] for x in write_cursor]
```

Comment:
in line 591 you are fetching all the tx hashes right? and in line 593 you are getting the customized ones and then you say 'Delete from history events where tx_hash in all and not in customized'. Can't you just do
```
ids_to_delete = list(set(all tx_hash) - set(customized))
```

and say

delete if in `ids_to_delete`. Also I'm not 100% sure but I believe doing a subquery is faster than checking if `event_identifier IN (?,....,?)`

Reply:
This is copy paste from the develop code that is tested. I would not change it. This should have been in the docstring.